assembly/src/ast/parsers/adv_ops.rs

```diff
@@ -77,6 +77,11 @@ pub fn parse_adv_inject(op: &Token) -> Result<Node, ParsingError> {
             }
             _ => return Err(ParsingError::extra_param(op)),
         },
+        "falcon_sign" => match op.num_parts() {
+            2 => AdvInject(FalconSign),
+            _ => return Err(ParsingError::extra_param(op)),
+        },
```

Comment:
If we introduce `SignatureKind` the instruction could be `adv.push_sig.rpo_falcon512` (the portion relevant for this match statement would be just `push_sig`).

Reply:
Done